airbyte-cdk/python/airbyte_cdk/sources/http_config.py

```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+# The goal of this variable is to make an implicit dependency explicit. As part of of the Concurrent CDK work, we are facing a situation
+# where the connection pool size is too small to serve all the threads (see https://github.com/airbytehq/airbyte/issues/32072). In
+# order to fix that, we will increase the requests library pool_maxsize. As there are many pieces of code that sets a requests.Session, we
+# are creating this variable here so that a change in one affects the other. This can be removed once we merge how we do HTTP requests in
```

Comment:
I know Joe worked on splitting the two. I vaguely remember the reason being that people use the HttpStream directly which is hard to modify the code while very few uses the HttpRequester directly. Splitting them allows for us to evolve the declarative one faster. 

I fear that as they diverge, the risk of forgetting to do a change in both and the complexity of it increases. Did we have a plan to merge the HTTP requesting part? Maybe this will be less relevant as we move things to concurrent and hence use more and more the declarative package?

Reply:
It's only a matter of prioritizing it. There aren't any real blockers to updating the HttpStream to depend on the HttpRequester (or other alternative to share the logic)